src/beacons/BeaconsContext.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import {useIsBeaconsEnabled} from './use-is-beacons-enabled';
+import {KETTLE_API_KEY} from '@env';
+import {Kettle} from 'react-native-kettle-module';
+import {NativeModules, Platform} from 'react-native';
+import {
+  BEACONS_CONSENTS,
+  allowedPermissionForKettle,
+  requestAndroidPermissions,
+} from './permissions';
+import {useBeaconsMessages} from './use-beacons-messages';
+import {storage} from '@atb/storage';
+
+type KettleInfo = {
+  isKettleStarted: boolean;
+  kettleIdentifier: string | null;
+  kettleConsents: Record<string, boolean> | null;
+  isBeaconsOnboarded: boolean;
+  privacyDashboardUrl?: string;
+  privacyTermsUrl?: string;
+};
+
+type BeaconsContextState = {
+  isKettleSDKInitialized: boolean;
+  isBeaconsSupported: boolean;
+  kettleInfo?: KettleInfo;
+  onboardForBeacons: () => void;
+  revokeBeacons: () => void;
+  deleteCollectedData: () => void;
+};
+
+const defaultState = {
+  isKettleSDKInitialized: false,
+  isBeaconsSupported: false,
+  kettleInfo: undefined,
+  onboardForBeacons: () => {},
+  revokeBeacons: () => {},
+  deleteCollectedData: () => {},
+};
+
+enum storeKey {
+  beaconsConsent = '@ATB_beacons_consent_granted',
+}
+
+const BeaconsContext = createContext<BeaconsContextState>(defaultState);
+
+const BeaconsContextProvider: React.FC = ({children}) => {
+  const [isBeaconsEnabled, debugOverrideReady] = useIsBeaconsEnabled();
+  const isBeaconsSupported =
+    isBeaconsEnabled && debugOverrideReady && !!KETTLE_API_KEY;
+  const {getRationaleMessages} = useBeaconsMessages();
+  const [isKettleSDKInitialized, setKettleSDKInitialized] = useState(false);
+  const [kettleInfo, setKettleInfo] = useState<KettleInfo>();
+
+  const updateKettleInfo = useCallback(async () => {
+    if (isBeaconsSupported && isKettleSDKInitialized) {
+      const status = await Kettle.isStarted();
+      const identifier = await Kettle.getIdentifier();
+      const consents = await Kettle.getGrantedConsents();
+      const privacyDashboardUrl = await Kettle.getPrivacyDashboardUrl();
+      const privacyTermsUrl = await Kettle.getPrivacyTermsUrl();
+
+      setKettleInfo({
+        isKettleStarted: status,
+        kettleIdentifier: identifier,
+        kettleConsents: consents,
+        isBeaconsOnboarded: Object.keys(consents).length > 0,
```

Comment:
Should it in addition check the `hasSeenShareTravelHabitsScreen` flag?

Reply:
It is to know if already the consent was granted, later `hasSeenShareTravelHabitsScreen` is used in combination with this `isBeaconsOnboarded` to know which announcement to show.

Reply:
Atm yes. I'm just thinking that it could be a good idea to make it explicit, so that the logic is directly linked. Not a big deal however..